File: android/src/main/java/com/getcapacitor/community/classes/events/MessageIDEvent.java
package com.getcapacitor.community.classes.events;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import java.util.UUID;

public class MessageIDEvent {

    @Nullable
    UUID messageID;

    public MessageIDEvent(@Nullable UUID messageID) {
        this.messageID = messageID;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();

        if (messageID != null) result.put("messageID", messageID.toString());

        return result;
    }
}

File: ios/Sources/BitchatPlugin/Classes/Events/MessageIDEvent.swift
import Foundation
import Capacitor

@objc public class MessageIDEvent: NSObject {
    let messageID: UUID?

    init(_ messageID: UUID?) {
        self.messageID = messageID
    }

    public func toJSObject() -> JSObject {
        var result = JSObject()

        if let messageID = messageID {
            result["messageID"] = messageID.uuidString.lowercased()
        }

        return result
    }
}
